test/template_literal.spec.js

```diff
@@ -52,170 +53,287 @@ _ddiast.plusOperator('<html lang="en">\\n    <body>\\n        <h1>XSS vulnerabil
+      rewriteAndExpect(js, '{\nconst c = a === "hello" ? _ddiast.plusOperator("World " + b, "World ", b) : "Moon";\n}')
+    })
 
-  it('typeof among variables is replaced by a variable', () => {
-    const js = 'const a = `He${typeof b}llo wor${a}ld`'
-    rewriteAndExpect(
-      js,
-      '{\nlet __datadog_test_0, __datadog_test_1, __datadog_test_2, __datadog_test_3;\n\
+    it('typeof among variables is replaced by a variable', () => {
+      const js = 'const a = `He${typeof b}llo wor${a}ld`'
+      rewriteAndExpect(
+        js,
+        '{\nlet __datadog_test_0, __datadog_test_1, __datadog_test_2, __datadog_test_3;\n\
         const a = (__datadog_test_3 = (__datadog_test_1 = typeof b, __datadog_test_2 = (__datadog_test_0 \
 = _ddiast.plusOperator(a + "ld", a, "ld"), _ddiast.plusOperator("llo wor" + __datadog_test_0, "llo wor", \
 __datadog_test_0)), _ddiast.plusOperator(__datadog_test_1 + __datadog_test_2, __datadog_test_1, __datadog_test_2))\
 , _ddiast.plusOperator("He" + __datadog_test_3, "He", __datadog_test_3));\n}'
-    )
-  })
+      )
+    })
 
-  it('tagged are not mofified', () => {
-    const js = 'const a = func`Hello${b}World`;'
-    rewriteAndExpectNoTransformation(js)
-  })
+    it('tagged are not mofified', () => {
+      const js = 'const a = func`Hello${b}World`;'
+      rewriteAndExpectNoTransformation(js)
+    })
 
-  it('tagged with child expressions are mofified', () => {
-    const js = 'const a = func`Hello${b + c}World`;'
-    rewriteAndExpect(js, '{\nconst a = func`Hello${_ddiast.plusOperator(b + c, b, c)}World`;\n}')
-  })
+    it('tagged with child expressions are mofified', () => {
+      const js = 'const a = func`Hello${b + c}World`;'
+      rewriteAndExpect(js, '{\nconst a = func`Hello${_ddiast.plusOperator(b + c, b, c)}World`;\n}')
+    })
 
-  it('nested template literal with +', () => {
-    const js = "const a = `Hello ${c} ${'how are u ' + `${'bye ' + d}`} world`;"
-    rewriteAndExpect(
-      js,
-      '{\nlet __datadog_test_0, __datadog_test_1, __datadog_test_2, __datadog_test_3, __datadog_test_4\
+    it('nested template literal with +', () => {
+      const js = "const a = `Hello ${c} ${'how are u ' + `${'bye ' + d}`} world`;"
+      rewriteAndExpect(
+        js,
+        '{\nlet __datadog_test_0, __datadog_test_1, __datadog_test_2, __datadog_test_3, __datadog_test_4\
 , __datadog_test_5, __datadog_test_6;\n\
 const a = (__datadog_test_6 = (__datadog_test_4 = c, __datadog_test_5 = (__datadog_test_3 = (__datadog_test_2 = (\
-__datadog_test_1 = (__datadog_test_0 = _ddiast.plusOperator(\'bye \' + d, \'bye \', d), _ddiast.plusOperator("" + \
-__datadog_test_0, "", __datadog_test_0)), _ddiast.plusOperator(\'how are u \' + __datadog_test_1, \'how are u \', \
+__datadog_test_1 = (__datadog_test_0 = _ddiast.plusOperator(\'bye \' + d, \'bye \', d), _ddiast.plusOperator(\
+__datadog_test_0 + "", __datadog_test_0, "")), _ddiast.plusOperator(\'how are u \' + __datadog_test_1, \'how are u \', \
```

Comment:
Small change in these lines, but nothing important, only the order from `"" + __datadog_test_0` to `__datadog_test_0 + ""`